lib/inferno/apps/cli/execute.rb

```diff
@@ -102,14 +104,28 @@ def outputter
         @outputter ||= OUTPUTTERS[options[:outputter]].new
       end
 
+      def load_preset_file_and_set_preset_id
```

Comment:
I find this very confusing and am having a really hard time wrapping my head around this method. Like why does this method call `apply_preset`, but that method isn't called anywhere if a preset id is used rather than a preset file?

Reply:
If the CLI is invoked with `--preset-file`, this method will load the file into `presets_repo` and reassign `options[:preset_id]` to the id in the file. Aside from that its just bolting down some error cases.

I removed the `test_session_repo#apply_preset` line which I forgot to do earlier. My understanding is `test_session_repo#apply_preset` just adds the inputs to `session_data_repo`, but `persist_inputs` already does that in the CLI implementation (to both preset and direct inputs).